src/backend/drm/atomic/mod.rs

```diff
@@ -0,0 +1,399 @@
+        add_props(&dev, res_handles.crtcs(), &mut old_state.1)?;
+        add_props(&dev, res_handles.framebuffers(), &mut old_state.2)?;
+        add_props(&dev, planes, &mut old_state.3)?;
+
+        map_props(&dev, res_handles.connectors(), &mut mapping.0)?;
+        map_props(&dev, res_handles.crtcs(), &mut mapping.1)?;
+        map_props(&dev, res_handles.framebuffers(), &mut mapping.2)?;
+        map_props(&dev, planes, &mut mapping.3)?;
+
+        dev.old_state = old_state;
+        dev.prop_mapping = mapping;
+        debug!(log, "Mapping: {:#?}", dev.prop_mapping);
+
+        Ok(AtomicDrmDevice {
+            dev: Rc::new(dev),
+            dev_id,
+            active,
+            backends: Rc::new(RefCell::new(HashMap::new())),
+            handler: None,
+            logger: log.clone(),
+        })
+    }
+}
+
+impl<A: AsRawFd + 'static> AsRawFd for AtomicDrmDevice<A> {
+    fn as_raw_fd(&self) -> RawFd {
+        self.dev.as_raw_fd()
+    }
+}
+
+impl<A: AsRawFd + 'static> BasicDevice for AtomicDrmDevice<A> {}
+impl<A: AsRawFd + 'static> ControlDevice for AtomicDrmDevice<A> {}
+
+impl<A: AsRawFd + 'static> Device for AtomicDrmDevice<A> {
+    type Surface = AtomicDrmSurface<A>;
+
+    fn device_id(&self) -> dev_t {
+        self.dev_id
+    }
+
+    fn set_handler(&mut self, handler: impl DeviceHandler<Device = Self> + 'static) {
+        self.handler = Some(RefCell::new(Box::new(handler)));
+    }
+
+    fn clear_handler(&mut self) {
+        let _ = self.handler.take();
+    }
+
+    fn create_surface(&mut self, crtc: crtc::Handle) -> Result<AtomicDrmSurface<A>, Error> {
+        if self.backends.borrow().contains_key(&crtc) {
+            return Err(Error::CrtcAlreadyInUse(crtc));
+        }
+
+        if !self.active.load(Ordering::SeqCst) {
+            return Err(Error::DeviceInactive);
+        }
+
+        let backend = Rc::new(AtomicDrmSurfaceInternal::new(
+            self.dev.clone(),
+            crtc,
+            self.logger.new(o!("crtc" => format!("{:?}", crtc))),
+        )?);
+
+        self.backends.borrow_mut().insert(crtc, Rc::downgrade(&backend));
+        Ok(AtomicDrmSurface(backend))
+    }
+
+    fn process_events(&mut self) {
+        match self.receive_events() {
+            Ok(events) => {
+                for event in events {
+                    if let Event::PageFlip(event) = event {
+                        trace!(self.logger, "Got event!");
+                        if self.active.load(Ordering::SeqCst) {
+                            if self
+                                .backends
+                                .borrow()
+                                .get(&event.crtc)
+                                .iter()
+                                .flat_map(|x| x.upgrade())
+                                .next()
+                                .is_some()
+                            {
+                                trace!(self.logger, "Handling event for backend {:?}", event.crtc);
```

Comment:
Same here, given we know the event is a page-flip, the message should say it.

Reply:
same